pkg/config/config.go

```
@@ -98,13 +99,11 @@ func WithArch(arch string) func(r *v1.Config) error {
 
 func NewConfig(opts ...GenericOptions) *v1.Config {
 	log := v1.NewLogger()
-	arch := func() string {
-		switch runtime.GOARCH {
-		case "amd64":
-			return "x86_64"
-		}
-		return runtime.GOARCH
-	}()
+	arch, err := utils.GolangArchToArch(runtime.GOARCH)
+	if err != nil {
+		return nil
```

should return error here?

---

Hmm, it only returns a `*v1.Config` but I added an error log to not swallow the error entirely!